source/framework/tools/src/TRestStringHelper.cxx

```diff
@@ -21,28 +21,56 @@ using namespace std;
 /// \brief Returns 1 only if valid mathematical expression keywords (or numbers)
 /// are found in the string **in**. If not it returns 0.
 ///
+/// By logic, mathematical expressions must have: +-*/e^% in the middle, or % in the end, or math functions in the beginning.
+/// despite those symbols, the string should be purely numeric.
+/// example: 
+/// 1+1 : expression
+/// sin(1.5) : expression
+/// 123456789 : not expression, It is a pure number that can be directly parsed.
 Int_t REST_StringHelper::isAExpression(string in) {
-    string temp = in;
-    vector<string> replace{"sqrt", "log", "exp", "gaus", "cos", "sin", "tan", "atan", "acos", "asin"};
-    for (int i = 0; i < replace.size(); i++) {
-        temp = Replace(temp, replace[i], "0", 0);
+    bool symbol = false;
+    bool numeric = false;
+    
+    if (in.length() < 2) // minimum expression: 3%
+        return 0;
+
+    vector<string> funcs{"sqrt", "log", "exp", "gaus", "cos", "sin", "tan", "atan", "acos", "asin"};
+    for (int i = 0; i < funcs.size(); i++) {
```

Comment:
for (const auto& item: funcs){
        if (in.find(item) != std::string::npos) {
            symbol = true;
            break;
        }
    }